src/Evolution/Systems/NewtonianEuler/BoundaryConditions/Reflection.hpp

```diff
@@ -0,0 +1,106 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <memory>
+#include <optional>
+#include <pup.h>
+#include <string>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Domain/BoundaryConditions/BoundaryCondition.hpp"
+#include "Evolution/BoundaryConditions/Type.hpp"
+#include "Evolution/Systems/NewtonianEuler/BoundaryConditions/BoundaryCondition.hpp"
+#include "Evolution/Systems/NewtonianEuler/Tags.hpp"
+#include "Options/Options.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+class DataVector;
+namespace gsl {
+template <class T>
+class not_null;
+}  // namespace gsl
+/// \endcond
+
+namespace NewtonianEuler::BoundaryConditions {
+/*!
+ * \brief Reflecting boundary conditions for Newtonian hydrodynamics.
+ *
+ * Ghost (exterior) data 'mirrors' interior volume data with respect to the
+ * boundary interface. i.e. reverts the normal component of velocity while using
+ * same values for other scalar quantities.
+ *
+ * \f{align*}
+ * v_\text{ghost}^i &= v_\text{int}^i - 2[(v^j-v_m^j)n_j]n^i
```

Comment:
I think the mesh velocity being in only the second term isn't very intuitive (at least not to me!). Could you add a sentence or two explaining that?

Reply:
Sure :)